src/apps/Permissions/Permissions.js

```diff
@@ -65,244 +35,177 @@ class Permissions extends React.Component {
-              ? resolveRole(location.app.proxyAddress, location.secondaryData)
-              : null
-
-          return (
-            <React.Fragment>
-              <AppLayout
-                title="Permissions"
-                navigationItems={navigationItems}
-                onNavigationBack={this.goToHome}
-                onMenuOpen={this.handleMenuPanelOpen}
-                smallViewPadding={30}
-                mainButton={{
-                  icon: <IconPlus />,
-                  label: 'Add permission',
-                  onClick: this.createPermission,
-                  disabled: appsLoading || permissionsLoading,
-                }}
-              >
-                <ScrollTopElement
-                  ref={el => {
-                    this._scrollTopElement = el
-                  }}
-                />
-
-                <Wrap>
-                  <Screen position={0} animate={animateScreens}>
-                    {location.screen === 'home' && (
-                      <Home
-                        apps={apps}
-                        appsLoading={appsLoading}
-                        permissionsLoading={permissionsLoading}
-                        onOpenApp={this.handleOpenApp}
-                        onOpenEntity={this.handleOpenEntity}
-                      />
-                    )}
-                  </Screen>
-
-                  <Screen position={1} animate={animateScreens}>
-                    {['app', 'entity'].includes(location.screen) && (
-                      <React.Fragment>
-                        {location.screen === 'app' && (
-                          <AppPermissions
-                            app={location.app}
-                            loading={appsLoading}
-                            address={location.address}
-                            onManageRole={this.handleManageRole}
-                          />
-                        )}
-                        {location.screen === 'entity' && (
-                          <EntityPermissions
-                            title="Permissions granted to this entity"
-                            loading={appsLoading || permissionsLoading}
-                            address={location.address}
-                          />
-                        )}
-                      </React.Fragment>
-                    )}
-                  </Screen>
-                </Wrap>
-              </AppLayout>
-
-              <AssignPermissionPanel
-                apps={apps}
-                opened={showAssignPermissionPanel}
-                onClose={this.closeAssignPermissionPanel}
-                wrapper={wrapper}
-              />
-
-              <ManageRolePanel
-                app={location.app}
-                apps={apps}
-                opened={managedRole !== null}
-                onClose={this.closeManageRolePanel}
-                role={managedRole}
+  }, [apps, params, onParamsRequest])
+
+  const location = getLocation(params, apps)
```

Comment:
No need to do anything, but you could've destructured `lcoation` instead of using it all around.